editor/header/tools/publish-button.js

```diff
@@ -8,81 +8,119 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from 'i18n';
+import { Component } from 'element';
 import { Button } from 'components';
 
 /**
  * Internal dependencies
  */
-import { editPost, savePost } from '../../actions';
+import {
+	editPost,
+	savePost,
+	successNotice,
+	errorNotice,
+} from '../../actions';
 import {
 	isSavingPost,
 	isEditedPostPublished,
 	isEditedPostBeingScheduled,
 	getEditedPostVisibility,
 	isEditedPostSaveable,
 	isEditedPostPublishable,
+	getCurrentPost,
+	didPostSaveRequestSucceed,
 } from '../../selectors';
 
-function PublishButton( {
-	isSaving,
-	isPublished,
-	onStatusChange,
-	onSave,
-	isBeingScheduled,
-	visibility,
-	isPublishable,
-	isSaveable,
-} ) {
-	const buttonEnabled = ! isSaving && isPublishable && isSaveable;
-	let buttonText;
-	if ( isPublished ) {
-		buttonText = __( 'Update' );
-	} else if ( isBeingScheduled ) {
-		buttonText = __( 'Schedule' );
-	} else {
-		buttonText = __( 'Publish' );
+class PublishButton extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onClick = this.onClick.bind( this );
 	}
-	let publishStatus = 'publish';
-	if ( isBeingScheduled ) {
-		publishStatus = 'future';
-	} else if ( visibility === 'private' ) {
-		publishStatus = 'private';
+
+	componentDidUpdate( prevProps ) {
```

Comment:
This is a good question.

> It sounds like just a side-effect of core editor actions, no matter who triggers the actions.

Maybe, it would make things easier to write for sure. If we see these actions like actions that could be triggered from different places (imagine other WP pages, for example a settings triggering bulk post update or something), this could cause bugs where notices are shown while we don't expect them or we want to make their content specific to the context.

We experienced this kind of issues with the site settings actions in Calypso.